### ISSUE TITLE
Improve DuckDB presence within IntelliJ IDEA

### ISSUE THREAD

**Issue by rsmckinney** [duckdb/duckdb-java#51]
As the screen capture shows, DuckDB is available as a data source within IntelliJ IDEA. However, DuckDB is in the [basic support](https://www.jetbrains.com/help/idea/other-databases.html) category, as opposed to the "complete support" category.

![image](https://github.com/duckdb/duckdb-java/assets/588843/07828bf0-dfec-4dba-88f6-ac8ff2f8174a)

This is a real disappointment when using DuckDB in IntelliJ in terms of SQL rendering, code completion, etc.

Even with the dialect set to PostgreSQL, queries that use extensions fail to parse in the editor. Usage of all those awesome extensions render in **red** and fail code completion. 

![image](https://github.com/duckdb/duckdb-java/assets/588843/add087b4-93fd-45a2-bc1a-52a3a4d32c31)

Would be nice to work with JetBrains to level up here, it would help DuckDB gain traction with Java.

**Comment by fourcadefran** [comment]
And Webstorm ❤️

**Comment by bsplosion** [comment]
With the launch of Quack protocol (https://duckdb.org/quack/) is there interest in re-engaging this ticket?  We've ended up leveraging Duckgres (https://github.com/PostHog/duckgres) plus a monumental amount of monkeypatching to get a fully Postgres-compliant remote service up and running for our developers to use for operations, but that's a hack and likely very brittle over time.  Getting first-class supported plugins out into the ecosystem would be really helpful.

**Comment by staticlibs** [comment]
@bsplosion 

Yes, we definitely would like to support more tools that use JDBC driver. Just the gap between "basic things work, data can be shown" and "tool is fully supported" can be pretty large. And every such tool needs to be addressed separately (example: #573).

In general, reworking the JDBC API data fetching on top of `DuckDBDataChunkReader` (#682) and improving support for extended types (`LIST`, `STRUCT` etc) in JDBC API/Appender/UDFs have higher priority.

**Comment by rsmckinney** [comment]
Looks promising, complete DuckDB support is on JetBrains roadmap: 
[jetbrains.com/issue/DBE-21385/DuckDB-support](https://youtrack.jetbrains.com/issue/DBE-21385/DuckDB-support#focus=Comments-27-13862438.0-0)

**Comment by geertjanw** [comment]
Easy fix, just switch over to the latest and greatest Apache NetBeans, (plus benefit from integration with Parquet and Excel). :-)

- https://github.com/geertjanw/Apache-NetBeans-Data-Wrangler
- https://foojay.io/today/netbeans-datawrangler-query-convert-and-edit-data-analytics-files/
